SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -198,14 +200,18 @@ void USpatialReceiver::HandleActorAuthority(Worker_AuthorityChangeOp& Op)
 				if (Op.authority == WORKER_AUTHORITY_AUTHORITATIVE)
 				{
 					Actor->Role = ROLE_Authority;
+					Actor->RemoteRole = ROLE_SimulatedProxy;
 
-					if (Actor->IsA<APawn>() || Actor->IsA<APlayerController>())
+					if (APlayerController* PlayerController = Cast<APlayerController>(Actor))
 					{
 						Actor->RemoteRole = ROLE_AutonomousProxy;
 					}
-					else
+					else if (APawn* Pawn = Cast<APawn>(Actor))
 					{
-						Actor->RemoteRole = ROLE_SimulatedProxy;
+						if (Pawn->IsPlayerControlled())
```

Comment:
@m-samiec, I realized this logic only matters on the server, when the player state already exists. So this check should be sufficient.